### PR TITLE
Fix compiled optimizer scheduler kwargs leak

### LINKED ISSUE #126131: Memory leak when using SequentialLR and ChainedLR schedulers
### 🐛 Describe the bug

Looks like memory is allocated in the caching allocator which doesn't get freed. I checked refcycle and it doesn't appear to be due to a cyclical reference in the python code. Repro below.

```py
import torch
from torch.optim.lr_scheduler import LambdaLR, ChainedScheduler, ConstantLR, SequentialLR
from torch.testing._internal.common_utils import CudaMemoryLeakCheck, TestCase, run_tests


LR_SCHEDULER_TO_KWARGS = {
    LambdaLR: {"lr_lambda": lambda x: 10},
    ChainedScheduler: {"schedulers": None},
    SequentialLR: {"schedulers": None, "milestones": [1, 2]},
    ConstantLR: {"factor": 0.001},
}


def create_scheduler(scheduler, optim):
    kwargs = LR_SCHEDULER_TO_KWARGS[scheduler]
    if "schedulers" in kwargs:
        kwargs["schedulers"] = [
            create_scheduler(ConstantLR, optim)
            for _ in range(2)
        ] + [create_scheduler(LambdaLR, optim)]

    if scheduler == ChainedScheduler:
        return scheduler(**kwargs)
    else:
        return scheduler(optim, **kwargs)

def chained_fn():
    from torch.testing._internal.common_utils import CudaMemoryLeakCheck
    with CudaMemoryLeakCheck(None,name="hi"):
        device="cuda:0"
        dtype=torch.float32
        optim_cls = torch.optim.ASGD
        kwargs = {'lr': 0.001, 'weight_decay': 0.1, 'maximize': True, 'capturable': True, 'foreach': False}
        scheduler_cls = ChainedScheduler
        print(scheduler_cls)
        print(kwargs)
        input = torch.ones([10, 10], device=device)
        model_eager = torch.nn.Sequential(
            *[torch.nn.Linear(10, 10, device=device) for _ in range(1)]
        )
        model_eager(input).sum().backward()
        opt_eager = optim_cls(model_eager.parameters(), **kwargs)
        scheduler_eager = create_scheduler(scheduler_cls, opt_eager)


def sequential_fn():
    from torch.testing._internal.common_utils import CudaMemoryLeakCheck
    with CudaMemoryLeakCheck(None,name="hi"):
        device="cuda:0"
        dtype=torch.float32
        optim_cls = torch.optim.ASGD
        kwargs = {'lr': 0.001, 'weight_decay': 0.1, 'maximize': True, 'capturable': True, 'foreach': False}
        scheduler_cls = SequentialLR
        print(scheduler_cls)
        print(kwargs)
        input = torch.ones([10, 10], device=device)
        model_eager = torch.nn.Sequential(
            *[torch.nn.Linear(10, 10, device=device) for _ in range(1)]
        )
        model_eager(input).sum().backward()
        opt_eager = optim_cls(model_eager.parameters(), **kwargs)
        scheduler_eager = create_scheduler(scheduler_cls, opt_eager)

# both sequential_fn and chained_fn fail
chained_fn()
```

### Versions

nightly:
a5c93a6899c657832944cd2eeb5069449e28dbea

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* (to be filled)

The compiled optimizer scheduler test helper reused the dict entries from
LR_SCHEDULER_TO_KWARGS directly. For SequentialLR and ChainedScheduler, the
helper fills in the "schedulers" entry with constructed child schedulers. That
mutated the global table and kept scheduler instances reachable after the test
body returned. The retained schedulers held the optimizer and model parameters,
which made CUDA memory leak checks report leaked allocations.

Copy the scheduler kwargs before filling nested schedulers so each scheduler
construction owns temporary kwargs. With the helper no longer retaining child
schedulers globally, SequentialLR and ChainedScheduler can be restored to the
compiled optimizer test matrix. Add a focused regression test that checks the
kwargs table is not mutated by create_scheduler.

Fixes #126131
Generated by my agent

Test Plan:
- python reproduction script from notes: pre-fix ChainedScheduler failed with CudaMemoryLeakCheck, post-fix ChainedScheduler and SequentialLR report warnings 0
- python test/inductor/test_compiled_optimizers.py CompiledOptimizerTests.test_create_scheduler_does_not_mutate_kwargs (blocked by unrelated local torchvision mismatch)
- python -m pytest -q test/inductor/test_compiled_optimizers.py::CompiledOptimizerTests::test_create_scheduler_does_not_mutate_kwargs (blocked by unrelated local torchvision mismatch)
- python runpy import-hook command for CompiledOptimizerTests.test_create_scheduler_does_not_mutate_kwargs: OK
- python runpy import-hook command for CompiledOptimizerTests.test_asgd_tensor_lr_weight_decay_maximize_capturable_cuda_chainedscheduler and ..._sequentiallr: OK
- lintrunner -a (failed on unrelated clang-tidy issues in torch/csrc/shim_common.cpp, torch/csrc/stable/accelerator.h, and torch/custom_class.h)
- lintrunner test/inductor/test_compiled_optimizers.py

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Review by mlazos — APPROVED** [review]

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
My agent says the failed Windows CPU jobs appear to be runner infrastructure failures: they die during import while constructing the Windows SSL context with ssl.SSLError [ASN1: NOT_ENOUGH_DATA], before this PR's scheduler test changes run. The merge workflow is still running due pending macOS jobs, so GitHub will not allow rerunning failed jobs yet; I will revisit after CI finishes.
